common/worker_versioning/worker_versioning.go

```diff
@@ -524,27 +526,73 @@ func ValidateVersioningOverride(override *workflowpb.VersioningOverride) error {
 	return nil
 }
 
-// FindDeploymentVersionForWorkflowID returns the deployment version that should be used for a
+// FindTargetDeploymentVersionAndRevisionNumberForWorkflowID returns the deployment version, revision number and a boolean
 // particular workflow ID based on the versioning info of the task queue. Nil means unversioned.
-func FindDeploymentVersionForWorkflowID(
+// The boolean is true if the target deployment is new, false if the target deployment is old.
+func FindTargetDeploymentVersionAndRevisionNumberForWorkflowID(
 	current *deploymentspb.DeploymentVersionData,
 	ramping *deploymentspb.DeploymentVersionData,
+	currentVersionRoutingConfig *deploymentpb.RoutingConfig,
+	rampingVersionRoutingConfig *deploymentpb.RoutingConfig,
 	workflowId string,
-) *deploymentspb.WorkerDeploymentVersion {
-	ramp := ramping.GetRampPercentage()
-	rampingVersion := ramping.GetVersion()
-	if ramp <= 0 {
+) (*deploymentspb.WorkerDeploymentVersion, int64) {
+	// Find most recent currentDeploymentVersion by comparing old vs new format timestamps
+	var finalCurrentDep *deploymentspb.WorkerDeploymentVersion
```

Comment:
Nit: better to call new vars based on the latest terminology: `finalCurrentVersion`